src/test/java/org/apache/mesos/scheduler/plan/DefaultPlanCoordinatorTest.java

```diff
@@ -0,0 +1,296 @@
+        return offers;
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testNoPlanManager() {
+        final DefaultPlanCoordinator coordinator = new DefaultPlanCoordinator(Arrays.asList(), planScheduler);
+        coordinator.processOffers(schedulerDriver, getOffers(1, 1, 1));
+    }
+
+    @Test
+    public void testOnePlanManagerPendingSufficientOffer() throws Exception {
+        when(stateStore.fetchTask(anyString())).thenReturn(Optional.empty());
+        final Plan plan = new DefaultPlanFactory(stateStore).getPlan(serviceSpecification);
+        final DefaultPlanCoordinator coordinator = new DefaultPlanCoordinator(
+                Arrays.asList(new DefaultPlanManager(plan, new DefaultStrategyFactory())),
+                planScheduler);
+        Assert.assertTrue(coordinator.processOffers(schedulerDriver, getOffers(2, 2000, 10000)).size() == 1);
+    }
+
+    @Test
+    public void testOnePlanManagerPendingInSufficientOffer() throws Exception {
+        when(stateStore.fetchTask(anyString())).thenReturn(Optional.empty());
+        final Plan plan = new DefaultPlanFactory(stateStore).getPlan(serviceSpecification);
+        final DefaultPlanCoordinator coordinator = new DefaultPlanCoordinator(
+                Arrays.asList(new DefaultPlanManager(plan, new DefaultStrategyFactory())),
+                planScheduler);
+        Assert.assertTrue(coordinator.processOffers(schedulerDriver, getOffers(2, 1, 1)).size() == 0);
+    }
+
+    @Test
+    public void testOnePlanManagerComplete() throws Exception {
+        when(stateStore.fetchTask(anyString())).thenReturn(Optional.empty());
+        final Plan plan = new DefaultPlanFactory(stateStore).getPlan(serviceSpecification);
+        plan.getPhases().get(0).getBlocks().get(0).forceComplete();
+        final DefaultPlanCoordinator coordinator = new DefaultPlanCoordinator(
+                Arrays.asList(new DefaultPlanManager(plan, new DefaultStrategyFactory())),
+                planScheduler);
+        Assert.assertTrue(coordinator.processOffers(schedulerDriver, getOffers(2, 2000, 10000)).size() == 0);
+    }
+
+    @Test
+    public void testTwoPlanManagersPendingPlansDisjointAssets() throws Exception {
+        when(stateStore.fetchTask(anyString())).thenReturn(Optional.empty());
+        final Plan planA = new DefaultPlanFactory(stateStore).getPlan(serviceSpecification);
+        final Plan planB = new DefaultPlanFactory(stateStore).getPlan(serviceSpecificationB);
+        final DefaultPlanManager planManagerA = new DefaultPlanManager(planA, new DefaultStrategyFactory());
+        final DefaultPlanManager planManagerB = new DefaultPlanManager(planB, new DefaultStrategyFactory());
+        final DefaultPlanCoordinator coordinator = new DefaultPlanCoordinator(
+                Arrays.asList(planManagerA, planManagerB), planScheduler);
+        Assert.assertEquals(2, coordinator.processOffers(schedulerDriver, getOffers(2, 2000, 10000)).size());
```

Comment:
I prefer this style of assertion, but please just choose one or the other.  You were using `assertTrue` on an equality expression before.